src/back/spv/writer.rs

```diff
@@ -108,13 +120,28 @@ impl Writer {
         arena: &crate::Arena<crate::Type>,
         handle: crate::Handle<crate::Type>,
     ) -> Word {
-        match self.lookup_type.lookup_id(handle) {
-            Some(word) => word,
-            None => {
-                let (instruction, id) = self.parse_type_declaration(arena, handle);
-                instruction.to_words(&mut self.logical_layout.declarations);
-                id
-            }
+        let ty = &arena[handle];
+
+        if let Some(id) = self
+            .lookup_type_new
+            .iter()
+            .find(|&(_, v)| match v {
+                LookupType::Handle(lookup_handle) => handle == *lookup_handle,
+                LookupType::Local(local_ty) => {
+                    match local_ty {
+                        LocalType::Scalar(ty_inner) => &ty.inner == ty_inner,
+                        // TODO Pointers
```

Comment:
Had an equivalent check for `LocalType::Pointer`, but it caused a weird behavior that a `OpTypePointer` referencing a result type of `OpTypePointer` would be created. Which is not allowed in SPIR-V.